app/templating/view_context.py

```diff
@@ -77,10 +76,10 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
                 answer_ids.append(answer['id'])
 
             if answer['type'] in ('Checkbox', 'Radio'):
-                options = answer['options']
-                context['form']['other_answer'][answer['id']] = []
-                for index in range(len(options)):
-                    context['form']['other_answer'][answer['id']].append(form.get_other_answer(answer['id'], index))
+                for option in answer['options']:
+                    if 'detail_answer' in option:
+                        context['form']['fields'][option['detail_answer']['id']] = \
+                            form.get_form_answer(option['detail_answer']['id'])
```

Comment:
Is this needed or can we append the detail answer id to `answer_ids` and it will be added to the context in the loop over answer ids below?